src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1782,6 +1782,17 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
                 return;
             }
 
+            // Note: reject requests whose client is not a known audience.
+            if (notification.Audiences.Contains(clientNotification.ClientId)) {
```

Comment:
Missing `!`? :smile: 

Note that this check won't work if the client was not authenticated.